src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/PathResolver/Error.hs

```diff
@@ -63,6 +72,18 @@ instance PrettyCodeAnn DependencyErrorGit where
       prefix :: Doc CodeAnn
       prefix = pretty @Text "Failed to obtain remote dependencies" <> line
 
+instance PrettyCodeAnn MissingLockfileDependency where
+  ppCodeAnn e =
+    "The dependency"
+      <+> code dependencyId
+      <+> "is declared in the package's juvix.yaml but is not declared in the lockfile:"
+      <+> pretty (e ^. missingLockfileDependencyPath)
+    where
+      dependencyId :: Doc CodeAnn
+      dependencyId = case e ^. missingLockfileDependencyDependency of
+        DependencyGit g -> pretty @Text (g ^. gitDependencyName)
+        DependencyPath p -> pretty @Text (show (p ^. pathDependencyPath))
+
```

Comment:
Let's make this error message more user-friendly by suggesting removing `juvix.lock.yaml` as it is out of sync.

> It appears that you need to manually remove the juvix.lock.yaml file and then run Juvix again.

Reply:
done

Reply:
NB: I'm adding an `update` command. So I'll change this error message to suggest that instead of removing the file when I raise that PR.